### PR TITLE
Support reading product dependencies from "META-INF/product-dependencies.json"

### PR DESCRIPTION
**Pull request by dansanduleac**
This will help API jars define appropriate an dependency recommendation range on the product that implements that API.

### CODE REVIEW

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy [hunk 1]
```diff
@@ -48,6 +49,7 @@ class CreateManifestTask extends DefaultTask {
     public static ObjectMapper jsonMapper = new ObjectMapper()
             .setSerializationInclusion(JsonInclude.Include.NON_NULL)
             .setPropertyNamingStrategy(PropertyNamingStrategy.KEBAB_CASE)
+    public static final String PDEP_FILE_PATH = "META-INF/pdep.json"
```
Comment (ferozco) [inline gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy]: pdeps.json

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy [hunk 2]
```diff
@@ -212,6 +197,53 @@ class CreateManifestTask extends DefaultTask {
...
+        return jsonMapper.readValue(pdeps, RecommendedProductDependencies)
+    }
+
+    @Nullable
+    RecommendedProductDependencies readProductDepsFromPdepFile(String coord, File file) {
+        try {
+            def zf = new ZipFile(file)
+            def entry = zf.getEntry(PDEPS_FILE_PATH)
+            if (entry == null) {
+                logger.debug("Pdeps file {} does not exist in jar for '{}'", PDEPS_FILE_PATH, coord)
```
Comment (uschi2000) [inline gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy]: what does "pdeps" stand for?

Comment (uschi2000) [reply gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy]: and where is it specified?

Comment (dansanduleac) [reply gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy]: product dependencies. Renamed the file & log lines to just say that, so it's obvious

Comment (uschi2000) [reply gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy]: cool

Comment (uschi2000) [reply gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy]: out of curiosity, why is a separate file preferred over the MANIFEST one?

**Review by ferozco — APPROVED** [review]

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy (hunk 2 above)
Comment (dansanduleac) [reply gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy]: It seemed cleaner to have a bespoke file that the conjure tooling would know how to generate.

**Pull request merged**